controllers/tests/cruisecontroloperation_controller_test.go

```diff
@@ -0,0 +1,454 @@
+			Eventually(func() bool {
+				operation1 := v1alpha1.CruiseControlOperation{}
+				err := k8sClient.Get(context.Background(), client.ObjectKey{
+					Namespace: kafkaCluster.Namespace,
+					Name:      opName1,
+				}, &operation1)
+				if err != nil {
+					return false
+				}
+				operation2 := v1alpha1.CruiseControlOperation{}
+				err = k8sClient.Get(context.Background(), client.ObjectKey{
+					Namespace: kafkaCluster.Namespace,
+					Name:      opName2,
+				}, &operation2)
+				if err != nil {
+					return false
+				}
+
+				return operation1.Status.NumberOfRetries == 0 && operation2.GetCurrentTaskState() == v1beta1.CruiseControlTaskCompleted
+			}, 40*time.Second, 500*time.Millisecond).Should(BeTrue())
+		})
+	})
+	When("there is a new remove_broker and an errored remove_broker operation with ignore ErrorPolicy", func() {
+		JustBeforeEach(func() {
+			cruiseControlOperationReconciler.Scaler = getScaleMock4(GinkgoT())
+			// Creating first operation
+			operation := generateCruiseControlOperation(opName1, namespace, kafkaCluster.GetName())
+			operation.Spec.ErrorPolicy = v1alpha1.ErrorPolicyIgnore
+			err := k8sClient.Create(context.TODO(), &operation)
+			Expect(err).NotTo(HaveOccurred())
+
+			operation.Status.CurrentTask = &v1alpha1.CruiseControlTask{
+				ID:        "12345",
+				Finished:  &metav1.Time{Time: time.Now().Add(-time.Second*v1alpha1.DefaultRetryBackOffDurationSec - 10)},
+				Operation: v1alpha1.OperationRemoveBroker,
+				State:     v1beta1.CruiseControlTaskCompletedWithError,
+			}
+			err = k8sClient.Status().Update(context.TODO(), &operation)
+			Expect(err).NotTo(HaveOccurred())
+			// Creating the second operation
+			operation = generateCruiseControlOperation(opName2, namespace, kafkaCluster.GetName())
+			err = k8sClient.Create(context.TODO(), &operation)
+			Expect(err).NotTo(HaveOccurred())
+			operation.Status.CurrentTask = &v1alpha1.CruiseControlTask{
+				Operation: v1alpha1.OperationRemoveBroker,
+			}
+			err = k8sClient.Status().Update(context.TODO(), &operation)
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("should execute the new remove_broker operation and should not execute the errored one", func() {
+			Eventually(func() bool {
+				operation1 := v1alpha1.CruiseControlOperation{}
+				err := k8sClient.Get(context.Background(), client.ObjectKey{
+					Namespace: kafkaCluster.Namespace,
+					Name:      opName1,
+				}, &operation1)
+				if err != nil {
+					return false
+				}
+				operation2 := v1alpha1.CruiseControlOperation{}
+				err = k8sClient.Get(context.Background(), client.ObjectKey{
+					Namespace: kafkaCluster.Namespace,
+					Name:      opName2,
+				}, &operation2)
+				if err != nil {
+					return false
+				}
+
+				return operation1.Status.NumberOfRetries == 0 && operation2.GetCurrentTaskState() == v1beta1.CruiseControlTaskCompleted
+			}, 40*time.Second, 500*time.Millisecond).Should(BeTrue())
+		})
+	})
+})
```

Comment:
**Required**: We **MUST** add a test case for 3 different type new operations, one of each in a non-priority ordering and expect to execute them in priority order to test the sorting and selection logic.

That would convince me of the current logic working as expected faster than words.

(Also we should write unit tests for those sort and select functions if we can.)

Reply:
It is hard to do this kind of test because I cannot add multiple cruiseControlOperations in the same time.  I rather add a unit test for this part.

Reply:
I am satisfied with the unit tests. thanks.